### PR TITLE
test: add tests

### LINKED ISSUE #18332: Add Ingest workflow checks HAPI tests
### Background

We've covered https://github.com/hiero-ledger/hiero-consensus-node/issues/18274 with unit tests.

The HAPI tests are dependent on https://github.com/hiero-ledger/hiero-consensus-node/issues/18273 since there we are changing the gRPC server to receive jumbo sized transactions. Otherwise, we will error out before the Ingest.

### Acceptance Criteria

1. Implement the happy tests
2. Implement the major edge cases

### Dependencies

1. #18273

### Definition of Ready (DoR) Checklist

- [x] Clear acceptance criteria
- [x] Clear and detailed description
- [x] Dependencies identified
- [ ] Links to documentation
- [x] Should be completable in 2-3 Days
- [ ] Initial draft of Low-level design document
- [x] At least high level test plan
- [ ] Groomed/Estimated

### Definition of Done (DoD) Checklist

- [ ] Acceptance Criteria complete
- [ ] No Codacy issues greater than minor (in new code)
- [ ] JavaDocs updated/created
- [ ] Code commented
- [ ] Unit tests created/updated
- [ ] 80% test code coverage (in new code)
- [ ] Happy Path and major negative cases in HAPI tests as applicable

### PR DESCRIPTION
**Pull request by vtronkov**
**Description**:
Add HAPI tests for #18312

**Related issue(s)**:

Fixes #18332

**Notes for reviewer**:
Note that #18312 needs to be merged and then these tests will pass

### CODE REVIEW

**Comment by mhess-swl** [comment]
Is this PR ready for review? It's still in draft mode

**Comment by vtronkov** [comment]
> Is this PR ready for review? It's still in draft mode

@mhess-swl Yes, it's ready for review. The reason for the draft mode is that it depends on #18312 in order for one of the tests to pass. When it's merged I will update and move it to ready for review. However, the two tests can still be reviewed even if they are not passing right now.

**Comment by mhess-swl** [comment]
@vtronkov ok great, thanks for clarifying!

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/ethereum/JumboTransactionsEnabledTest.java [hunk 1]
```diff
@@ -67,4 +68,51 @@ public Stream<DynamicTest> jumboTransactionShouldPass() {
...
+                        .signingWith(SECP_256K1_SOURCE_KEY)
+                        .payingWith(RELAYER)
+                        .gasLimit(1_000_000L)
+                        .hasPrecheck(TRANSACTION_OVERSIZE));
+    }
+
+    @HapiTest
+    @DisplayName("Ethereum contract create jumbo transaction more then 6kb smart contract should pass")
+    public Stream<DynamicTest> ethereumContractCreateJumboTxnMoreThen6Kb() {
+        final var contract = "TokenCreateContract";
```
Comment (mhess-swl) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/ethereum/JumboTransactionsEnabledTest.java:99]: I don't see this contract added in this PR, so it's an existing contract? (Maybe a big system contract?)

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/ethereum/JumboTransactionsEnabledTest.java [hunk 2]
```diff
@@ -67,4 +68,51 @@ public Stream<DynamicTest> jumboTransactionShouldPass() {
...
+                        .markAsJumboTxn()
+                        .type(EthTxData.EthTransactionType.EIP1559)
+                        .signingWith(SECP_256K1_SOURCE_KEY)
+                        .payingWith(RELAYER)
+                        .gasLimit(1_000_000L)
+                        .hasPrecheck(TRANSACTION_OVERSIZE));
+    }
+
+    @HapiTest
+    @DisplayName("Ethereum contract create jumbo transaction more then 6kb smart contract should pass")
```
Comment (mhess-swl) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/ethereum/JumboTransactionsEnabledTest.java:97]:
```suggestion
    @DisplayName("Allows Ethereum jumbo contract create jumbo above max transaction size of 6kb")
```

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/ethereum/JumboTransactionsEnabledTest.java (hunk 1 above)
Comment (vtronkov) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/ethereum/JumboTransactionsEnabledTest.java:99]: Yes, it's not a new contract. It's just a random contract that is bigger than 6kb.

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/ethereum/JumboTransactionsEnabledTest.java (hunk 2 above)
Comment (vtronkov) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/ethereum/JumboTransactionsEnabledTest.java:97]: Changed

**Review by mhess-swl — DISMISSED** [review]

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/ethereum/JumboTransactionsEnabledTest.java [hunk 3]
```diff
@@ -75,6 +75,21 @@ public Stream<DynamicTest> jumboTransactionDisabled() {
...
+        return hapiTest(
+                // The feature flag is only used once at startup (when building gRPC ServiceDefinitions),
+                // so we can't toggle it via overriding(). Instead, we need to upgrade to the config version.
+                prepareFakeUpgrade(),
+                upgradeToNextConfigVersion(
+                        Map.of(
+                                "jumboTransactions.isEnabled", "true",
+                                "hedera.transaction.maxMemoUtf8Bytes", "10000"),
+                        noOp()));
+    }
```
[the remark this answers is not in the document]
Comment (vtronkov) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/ethereum/JumboTransactionsEnabledTest.java:89]: Okay, I tested it locally, and it's not leaking.
Here is the setup:
I created a package with two test classes:
1. I had the `updateTheConfig` test inside
2. I had the `jumboTransactionDisabled`

Because the `updateTheConfig` test didn't leak, both tests passed:
image
